src/api/c/random.cpp

```diff
@@ -110,13 +106,14 @@ static void validateRandomType(const af_random_engine_type type) {
         AF_ERROR("Invalid random type", AF_ERR_ARG);
     }
 }
+}
 
 af_err af_get_default_random_engine(af_random_engine *r) {
     try {
         AF_CHECK(af_init());
 
-        thread_local RandomEngine re;
-        *r = static_cast<af_random_engine>(&re);
+        thread_local RandomEngine *re = new RandomEngine;
+        *r = static_cast<af_random_engine>(re);
```

Comment:
Why are we creating a intentional leak here now ?

Reply:
Yeah, its causing problems on Windows. Sometimes the destructor is called after the CUDA driver is unloaded and it causes segfaults on exit.